### PR TITLE
CredoBear.py: Adds CredoBear [WIP]

### PR DESCRIPTION
**Pull request by palash25**
Adds CredoBear to lint Elixir files

Closes: https://github.com/coala/coala-bears/issues/1266

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by shreyateeza** [comment]
@palash25 Check the items in the checklist.

**Comment by sangamcse** [comment]
Change your commit message. Just remove the 3rd person "s".

> CredoBear.py: Adds CredoBear

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
@gitmate-bot rebase

File: .ci/deps.sh [hunk 1]
```diff
@@ -62,3 +62,24 @@ wget "https://downloads.sourceforge.net/project/astyle/astyle/astyle%203.0.1/ast
 tar -xvzf ~/astyle.tar.gz -C ~/
 make -C ~/astyle/build/gcc
 sudo make install -C ~/astyle/build/gcc
+
+# Elixir commands
+wget https://packages.erlang-solutions.com/erlang-solutions_1.0_all.deb
```
Comment (jayvdb) [inline .ci/deps.sh]: can we add a repo instead, and let apt-get do its trick elsewhere?

If not, at least move this out to a separate `deps.mix.sh`

File: .travis.yml [hunk 2]
```diff
@@ -88,7 +89,7 @@ env:
     - R_LIB_USER=~/R/Library
     - LINTR_COMMENT_BOT=false
     - PATH="/opt/cabal/1.24/bin:$PATH:$TRAVIS_BUILD_DIR/node_modules/.bin:$TRAVIS_BUILD_DIR/vendor/bin:$HOME/dart-sdk/bin:$HOME/.cabal/bin:$HOME/infer-linux64-v0.7.0/infer/bin:$HOME/pmd-bin-5.4.1/bin:$HOME/bakalint-0.4.0:$HOME/elm-format-0.18:$HOME/.local/tailor/tailor-latest/bin:$HOME/phpmd"
-
```
Comment (jayvdb) [inline .travis.yml]: leave the blank line alone; it never hurt you.

File: .travis.yml [hunk 3]
```diff
@@ -88,7 +89,7 @@ env:
...
+    - cd ~/coala-bears/tests/elixir && mix deps.get && mix compile
```
Comment (jayvdb) [inline .travis.yml]: this doesnt look like a env variable ... ?

File: bears/elixir/CredoBear.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+    @classmethod
+    def check_prerequisites(cls):  # pragma: no cover
+        """
+        Checks if ``mix`` and ``bash`` are installed
+        :return:
+        ``True`` if ``mix``, ``bash`` and ``credo``
+        are installed otherwise ``False``
+        """
+        if which('bash') is None:
```
Comment (jayvdb) [inline bears/elixir/CredoBear.py]: eww, no thx.

File: bears/elixir/CredoBear.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            return 'bash is not installed.'
+        if which('mix') is None:
```
Comment (jayvdb) [inline bears/elixir/CredoBear.py]: add a `Requirement` for this.

File: bears/elixir/CredoBear.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+                                    start_line=line,
+                                    start_column=column))
+
+        yield Result(
+                self, message_map[severity],
+                affected_code, severity_map[severity],
+                additional_info=(msg))
+
+    @staticmethod
+    def create_arguments(self, filename, file, lintmode: str, config_file):
```
Comment (jayvdb) [inline bears/elixir/CredoBear.py]: `lintmode: str = 'flycheck'` ?

and re-position `config_file`

File: tests/elixir/mix.exs [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  def application do
+    [extra_applications: [:logger]]
+  end
+
+  defp deps do
+    [{:credo, "~> 0.5", only: [:dev, :test]}]
+  end
+end
```
Comment (jayvdb) [inline tests/elixir/mix.exs]: EOL at EOF

**Pull request closed without merging**